src/irmin/conf.ml

```diff
@@ -83,9 +83,13 @@ let conv t = t.conv
 let default t = t.default
 
 let key ?docs ?docv ?doc name conv default =
-  let to_univ, of_univ = Univ.create () in
-  let id = Oo.id (object end) in
-  { id; to_univ; of_univ; name; docs; docv; doc; conv; default }
+  let re = Str.regexp ".*[^0-9a-z_-].*" in
+  let illegal = Str.string_match re name 0 in
```

Comment:
NB: `Str` is not part of any MirageOS unikernel (since https://github.com/mirage/mirage/commit/461ba1083756448f9a6ba97c06cfe8ad0e3d7810), motivation behind that: https://mirage.io/blog/ocaml-regexp

Reply:
I didn't know that (though I suspected it...), that's fixed !